compiler/rustc_middle/src/thir.rs

```diff
@@ -375,6 +375,13 @@ pub enum ExprKind<'tcx> {
     Loop {
         body: ExprId,
     },
+    /// A `#[loop_match] loop { state = 'blk: { match state { ... } } }` expression.
+    LoopMatch {
+        state: ExprId,
+
+        region_scope: region::Scope,
+        arms: Box<[ArmId]>,
+    },
```

Comment:
Please add the relevant `pretty` test for this and the other to ensure these get pretty printed correctly.

Reply:
what sort of a test should I add for this? would `//@ pretty-mode:hir,typed` exercise the relevant logic (we only add this node for `thir`, not for `hir`).

Reply:
We don't have pretty printing support for THIR. Only `-Zunpretty=thir-tree` and `-Zunpretty=thir-flat` exist, which both use `Debug` impls.